### ISSUE TITLE
Cookie support broken in Electron 5.0.x?

### ISSUE THREAD

**Issue by danawoodman** [electron/electron#18220]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:** `5.0.1`
* **Operating System:** `mac OSX 10.14.4`
* **Last Known Working Electron version:**: `4.x`

### Expected Behavior

I should be able to set/remove a cookie.

### Actual Behavior

I cannot set or remove a cookie

### To Reproduce

```js
import { session } from 'electron'

async function testCookies() {
  if (!session || !session.defaultSession)
    throw new Error('Could not get default user session')

  const cookies = session.defaultSession.cookies
  const name = 'foo'
  const url = '/'

  await cookies.set({
    url,
    name,
    value: 'Hello World!',
    expirationDate: 99999999.0,
  })

  console.log('COOKIES BEFORE:', await cookies.get({}))

  await cookies.remove(url, name)

  console.log('COOKIES AFTER:', await cookies.get({}))
}

//....
app.on('ready', () => {
  testCookies()
})

```

### Additional Information

Running this on a Mac and have tried many permutations of the above but can't get anything to work. Things seemed to work in 4.x

**Comment by grantcv1** [comment]
I seem to be having similar problems with localStorage. It's now cleared between sessions

**Comment by thesuperhomie** [comment]
@danawoodman & @grantcv1 either of you guys want to take this on? If not I'll grab it :)

**Comment by erickzhao** [comment]
Heya @danawoodman,

@DeerMichel and I investigated further into this issue, but couldn't find anything wrong with cookie support in `v5.0.3`

[Here's a link to a gist](https://gist.github.com/1554a78fda2124e34000692291e73521) you can paste into [Electron Fiddle](https://electronjs.org/fiddle). This contains a minimum working example for the Cookies API (also available under `Show Me -> Electron APIs -> Cookies` in the Fiddle menu bar).

Please let us know if the below comments don't solve your issue.

-------------------------

After some investigation, there seems to be a problem in the parameters from your posted example: the `expirationDate` option uses Unix epoch time, so `expirationDate: 99999999.0` would tell your cookie to expire on Saturday, March 3, 1973. This would instantly delete it.

-------------------------

Also, setting `url: '/'` as an option leads to silent failure in Electron < 6. It should be rejecting the promise instead of resolving it, as the cookie domain associated to the URL is invalid. In this case, the invalid cookie will be unable to be filtered using the `cookie.get({ url })` nor removed using `cookie.remove({ url })`.

This is a bug on Electron's end, and a fix was proposed in #18756 to prevent invalid cookies from being generated for Electron <=5. This problem has been fixed already as of the Electron 6 beta, as the cookie error handling logic was improved. 

--------------------------

Putting it all together, [here is a working gist](https://gist.github.com/erickzhao/921b377a87e57ef85920d14efa2f0d43) for Electron Fiddle with some modifications to your original code.

**Comment by codebytere** [comment]
Closing as a result of @erickzhao's and @DeerMichel's investigation above.
